libraries/clickhouse-helpers/clickhouse_helpers/execute_sql/test_execute_sql.py

```diff
@@ -40,32 +40,20 @@ def test_insert_dataframe_works_regardless_of_column_order(db_context: DBContext
         "test_table",
         pd.DataFrame(
             {"col1": [3, 4], "col2": ["a", "b"], "id": [1, 2], "col3": ["c", "d"]}
-        ).set_index("id"),
+        ),
     )
 
     res = query_dataframe(db_context, "SELECT * FROM test_table")
     assert (res.values == TEST_DF.values).all()
 
 
-def test_insert_dataframe_works_with_missing_columns(db_context: DBContext):
-    insert_dataframe(
-        db_context,
-        "test_table",
-        pd.DataFrame({"col2": ["a", "b"], "id": [1, 2]}).set_index("id"),
-    )
-
-    res = query_dataframe(db_context, "SELECT * FROM test_table")
-    print(res.values)
-
-    # Note, if value is not Nullable table definition,
-    # missing strings are cast to empty string and missing ints are cast to 0
-
-    assert (
-        res.values
-        == pd.DataFrame(
-            {"id": [1, 2], "col1": [0, 0], "col2": ["a", "b"], "col3": ["", ""]}
-        ).values
-    ).all()
+def test_insert_throws_on_with_missing_columns(db_context: DBContext):
```

Comment:
The new insert method for pandas dataframes does not do any magic on missing columns but tells you to provide them all and make inputation explicit. Which I actually like.

Reply:
This is great!